socorro/external/postgresql/setupdb_app.py

```diff
@@ -268,75 +290,88 @@ def create_table(element, compiler, **kw):
                 print 'Only 9.2+ is supported at this time'
                 return 1
 
-            connection = db.engine.connect()
-            if self.config.get('dropdb'):
-                if 'test' not in self.database_name and not self.force:
+        # We can only do the following if the DB is not Heroku
+        # XXX Might add the heroku commands for resetting a DB here
+        if self.config.dropdb and not self.on_heroku:
+            with PostgreSQLAlchemyManager(connection_url, self.config.logger,
+                                          autocommit=False) as db:
+                if 'test' not in self.database_name and not self.config.force:
                     confirm = raw_input(
                         'drop database %s [y/N]: ' % self.database_name)
                     if not confirm == "y":
                         logging.warn('NOT dropping table')
                         return 2
+                db.drop_database(self.database_name)
+
+        if self.config.createdb:
+            with PostgreSQLAlchemyManager(connection_url, self.config.logger,
+                                          autocommit=False) as db:
+                db.create_database(self.database_name)
+                db.create_roles(self.config)
```

Comment:
with autocommit set to False, why don't we commit at the end?  when the connection is closed at the end of the context-manager section, does it automatically commit or rollback?